nixos/modules/services/networking/sozu.nix

```diff
@@ -0,0 +1,479 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sozu;
+  format = pkgs.formats.toml {};
+
+  globalSettings = format.generate "global.toml" cfg.global;
+  optionalSettings = format.generate "optional.toml" cfg.optional;
+in
+{
+  options.services.sozu = {
+    enable = mkEnableOption "Sozu server";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.sozu;
+      description = "Sozu package to use.";
+    };
+
+    configFile = mkOption {
```

Comment:
Are there configurations which aren't possible with `settings`, or is this for convenience to hand write your own config?

Reply:
It's for convenience.